src/engine/RuntimeInformation.h

```diff
@@ -15,7 +16,18 @@
 
 class RuntimeInformation {
  public:
-  friend void to_json(nlohmann::json& j, const RuntimeInformation& rti);
+  // The following declarations introduce an ordered_json type
+  // that functions exactly like nlohmann::json but keeps
+  // object keys in insertion order.
+  // This helps to make the JSON more readable as it forces children to be
+  // printed after their parents. See also
+  // https://github.com/nlohmann/json/issues/485#issuecomment-333652309
+  template <class K, class V, class dummy_compare, class A>
+  using fifo_map = nlohmann::fifo_map<K, V, nlohmann::fifo_map_compare<K>, A>;
+  using ordered_json = nlohmann::basic_json<fifo_map>;
```

Comment:
It seems to work, but just out of curiosity: how is `ordered_json` a complete type here?
As I understand it `fifo_map` is still templated on `<K, V, A>` and I don't see an instantiation,
but later on we can write `void foo(RuntimeInformation::ordered_json&j)` without running into trouble.

Reply:
I think while `fifo_map` is indeed still templated and couldn't be used without template arguments, `ordered_json` is `nlohmann::basic_json<fifo_map>` which determines the `K` and `V` arguments.